docs/misc/components.md

```diff
@@ -0,0 +1,105 @@
+* `Component#selector(String pattern, Optional<Component> separator)` - creates component for displaying names of entities selected by [entity selector][selectors] `pattern`.
+
+Applying Style
+==============
+
+`Style` specifies how component should look like, such as bold, italic, color, custom font, etc.
+Style is immutable objects and any modification to already existing style will instead create new instance. Its constructor is `private` so you'll have to copy and reconfigure `Style.EMPTY`.
+
+Style's default configurations are `null` instead of concrete value. This means not explicitly specified configuration may be overridden.
+After configuring style to your preference, you can apply it to your component with either `MutableComponent#setStyle` for overwriting previous style, or `#withStyle` for only applying non-null configurations to already existing style.
+
+Here's an example of how you can stylize your component:
+```java
+MutableComponent text = Component.literal("Hello!"); // Creates MutableComponent wrapping literal "Hello!"
+
+Style blueItalic = Style.EMPTY // Copies empty style and sets color to blue and makes it italic
+        .withColor(0x0000FF)
+        .withItalic(true);
+Style red = Style.EMPTY // Copies empty style and sets color to red
+        .withColor(0xFF0000);
+Style bold = Style.EMPTY // Copies empty style and makes it bold
+        .withBold(true);
+
+Style doubleLines = Style.EMPTY // Copies empty style and makes it both underlined and strikethrough
+        .withUnderlined(true)
+        .withStrikethrough(true);
+
+text.setStyle(blue); // Sets style of the text to be blue and italic
+text.withStyle(red).withStyle(bold).withStyle(doubleLines); // Overwrites blue and italic style to be red, bold, underlined, and strikethrough
+```
+This results in red, bold text with two lines.
+
+Chaining Components
+===================
+
+`MutableComponent` can have additional components chained as siblings with `MutableComponent#append`. Chained components can be retrieved with `MutableComponent#getSiblings`.
+
+`Component` stores its siblings like a tree, and its style is inherited to its siblings.
+Below example will progressively apply style on top of each other:
+```java
+Style bold = Style.EMPTY.withBold(true);
+Style italic = Style.EMPTY.withItalic(true);
+Style strike = Style.EMPTY.withStrikethrough(true);
+Style underlined = Style.EMPTY.withUnderlined(true);
+
+MutableComponent a = Component.literal("a");
+MutableComponent b = Component.literal("b").setStyle(bold);
+MutableComponent c = Component.literal("c").setStyle(italic);
+MutableComponent d = Component.literal("d").setStyle(strike);
+MutableComponent e = Component.literal("e").setStyle(underlined);
+a.append(b);
+b.append(c);
+c.append(d);
+d.append(e);
+```
+[Here's how it looks like in-game][style_annotated]
```

Comment:
I would use a better image. You can get one using a command block and title text or draw a black ground and render the component on top.